ansible/roles/app_submitty/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+    state: directory
+    mode: "0755"
+
+# Change version to match current release
+- name: Checkout specific version of the submitty git repo
+  ansible.builtin.git:
+    repo: https://github.com/Submitty/Submitty.git
+    dest: /usr/local/submitty/GIT_CHECKOUT/Submitty
+    version: v24.01.01
+    force: yes
+    update: yes
+
+# this script only uses PAM and email server
+- name: Run Submitty install script.
+  ansible.builtin.shell:
+    executable: "/bin/bash"
+    cmd: |
+      set -o pipefail
+      echo "localhost
+        5432
+        "{{ submitty_dbuser }}"
+        "{{ submitty_dbuser_password }}"
+        "{{ submitty_course_dbuser }}"
+        "{{ submitty_course_dbuser_password }}"
+        "{{ timezone }}"
+        "{{ language }}"
+        "https://{{ submitty_url }}"
+        "{{ vcs_url }}"
+        "{{ institution_name }}"
+        "{{ sysadmin_email }}"
+        "{{ submitty_email }}"
+        "{{ institution_url }}"
+        1
+        "{{ admin_email }}"
+        y
+        "{{ admin_email }}"
+        "{{ admin_password }}"
+        "{{ submitty_ip }}"
+        "{{ submitty_email }}"
+        "{{ noreply_email }}"
+        "{{ smtp_server }}"
+        "{{ smtp_port }}"
+        "{{ institution_url }}"
+        " | bash -p ./.setup/install_system.sh > /usr/local/submitty/install.log 2>&1
+  args:
+    chdir: /usr/local/submitty/GIT_CHECKOUT/Submitty
+  become: yes
+  become_user: root
+
+- name: check to see if installation succeeded
+  ansible.builtin.shell: grep "INSTALLATION SUCCESS!" /usr/local/submitty/install.log
+
+- name: Update the PHP disabled functions
+  ansible.builtin.lineinfile:
+    path: /etc/php/8.1/fpm/php.ini
+    regexp: "^disable_functions = "
+    line: disable_functions = popen,pclose,proc_open,php_real_logo_guid,php_egg_logo_guid,php_ini_scanned_files,php_ini_loaded_file,readlink,symlink,link,set_file_buffer,proc_close,proc_terminate,proc_get_status,proc_nice,getmyuid,getmygid,getmyinode,putenv,get_current_user,magic_quotes_runtime,set_magic_quotes_runtime,import_request_variables,ini_alter,stream_socket_server,stream_socket_accept,stream_socket_pair,stream_get_transports,stream_wrapper_restore,mb_send_mail,openlog,syslog,closelog,pfsockopen,posix_kill,apache_child_terminate,apache_get_modules,apache_get_version,apache_lookup_uri,apache_reset_timeout,apache_response_headers,virtual,system,phpinfo,exec,shell_exec,passthru,
+
+- name: Set up Apache Submitty configuration
+  ansible.builtin.copy:
+    src: /usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/apache/submitty.conf
+    dest: /etc/apache2/sites-available/submitty.conf
+    remote_src: yes
+    mode: "0644"
+
+- name: Set up SSL in Apache Submitty configuration
+  ansible.builtin.replace:
+    path: /etc/apache2/sites-available/submitty.conf
+    regexp: '<VirtualHost \*:80>'
+    replace: "{{ lookup('file', 'append_to_submitty.conf') }}"
+  when: ssl_enabled | bool
+
+- name: Add ip to the Apache Submitty configuration
```

Comment:
Why is the host being changed to ip address? ( I think this task could be removed.)

Reply:
I believe we are changing the host to the ip address in order to access the web interface. Do we still want to remove this task.